src/Actions/StockPosting/CreateStockPostingsFromOrder.php

```diff
@@ -0,0 +1,261 @@
+            ->with([
+                'orderPositions' => fn ($query) => $query->where('is_free_text', false)
+                    ->select(['id', 'order_id', 'product_id', 'warehouse_id', 'amount']),
+                'orderPositions.reservedStock',
+                'orderPositions.stockPostings',
+                'orderPositions.product:id,name,is_nos',
+                'orderType:id,order_type_enum',
+            ])
+            ->first(['id', 'order_type_id', 'order_number']);
+
+        $postStock = ! data_get($this->data, 'only_reserve_stock', false);
+
+        $multiplier = $order->orderType->order_type_enum->multiplier();
+        foreach ($order->orderPositions as $orderPosition) {
+            $posting = bcabs($orderPosition->stockPostings()->sum('posting'));
+
+            if (bccomp($posting, $orderPosition->amount) >= 0) {
+                continue;
+            }
+
+            $orderPosition->warehouse_id ??= resolve_static(Warehouse::class, 'default')?->id;
+            $open = bcsub($orderPosition->amount, $posting);
+
+            // Handle Purchase Orders and alike.
+            if ($multiplier === -1 && $postStock) {
+                CreateStockPosting::make([
+                    'warehouse_id' => $orderPosition->warehouse_id,
+                    'product_id' => $orderPosition->product_id,
+                    'order_position_id' => $orderPosition->id,
+                    'purchase_price' => $orderPosition->unit_net_price,
+                    'posting' => $open,
+                    'description' => $order->order_number,
+                ])
+                    ->checkPermission()
+                    ->validate()
+                    ->execute();
+
+                continue;
+            } elseif ($multiplier === -1) {
+                continue;
+            }
+
+            // Handle Sales Orders and alike.
+            $reserved = $orderPosition->reservedStock()->sum('reserved_amount');
+
+            // Check if stock is already reserved.
+            if (bccomp($open, $reserved) <= 0) {
+                if ($postStock) {
+                    $this->postReservedStock($orderPosition, $open, $order->order_number);
+                }
+
+                continue;
+            }
+
+            $availableStock = resolve_static(StockPosting::class, 'query')
+                ->where('product_id', $orderPosition->product_id)
+                ->where('warehouse_id', $orderPosition->warehouse_id)
+                ->sum('remaining_stock');
+
+            if (bccomp($open, bcadd($availableStock, $reserved)) === 1 && ! $orderPosition->product->is_nos) {
+                throw ValidationException::withMessages([
+                    'orderPositions' => [
+                        __(
+                            'Not enough stock available in warehouse :warehouse for product :product.',
+                            [
+                                'warehouse' => $orderPosition->warehouse->name,
```

Comment:
Attempt to read property "name" on null. didnt investigate why warehouse is null, just tried to use more that available stock